pkg/apitype/events.go

```diff
@@ -94,6 +112,8 @@ type StepEventMetadata struct {
 	Keys []string `json:"keys,omitempty"`
 	// Keys that changed with this step.
 	Diffs []string `json:"diffs,omitempty"`
+	// The diff for this step as a list of full paths and difference types.
+	RichDiff map[string]DiffKind `json:"richDiff,omitempty"`
```

Comment:
Note: I would love to find a better name for this thing. `propertyDiffs`, `detailedDiff`, etc.

Reply:
DetailedDiff > RichDiff IMO

Reply:
it's not clear to me what `full paths` means.

Reply:
I am going to move forward with `DetailedDiff`.